cadc-util/src/main/java/ca/nrc/cadc/auth/AuthenticationUtil.java

```diff
@@ -138,76 +134,70 @@ public class AuthenticationUtil {
-            Object o = c.newInstance();
-            Authenticator ret = (Authenticator) o;
-            log.debug("Authenticator: " + cname);
-            return ret;
-        } catch (Throwable t) {
-            if (!defaultImplClass.equals(cname) || c != null) {
-                log.error("failed to load Authenticator: " + cname, t);
-            }
-            log.debug("failed to load Authenticator: " + cname, t);
-        }
-        log.debug("Authenticator: null");
-        return null;
-    }
-
     /**
      * Load the available IdentityManager implementation. This utility method will
      * check the <code>ca.nrc.cadc.auth.IdentityManager</code> system property for a
-     * configured class name (default class name:
-     * <code>ca.nrc.cadc.auth.IdentityManagerImpl</code>). The easiest way for
-     * software implementers to customize behavior is to create their own
-     * IdentityManagerimpl class and add it to the classpath.
+     * configured class name. If not configured or loading the configured implementation
+     * fails, the default is the <code>ca.nrc.cadc.auth.NoOpIdentityManager</code>.
      * 
-     * @return an IdentityManager implementation or null if none provided
+     * @return an IdentityManager implementation
      */
     public static IdentityManager getIdentityManager() {
-        String defaultImplClass = IdentityManager.class.getName() + "Impl";
         String cname = System.getProperty(IdentityManager.class.getName());
-        Class c = null;
-        if (cname == null) {
-            cname = defaultImplClass;
+        if (cname != null) {
+            try {
+                Class c = Class.forName(cname);
+                Object o = c.getConstructor().newInstance();
+                IdentityManager ret = (IdentityManager) o;
+                log.debug("IdentityManager: " + cname);
+                return ret;
+            } catch (ClassNotFoundException 
+                    | IllegalAccessException | IllegalArgumentException | InstantiationException 
+                    | NoSuchMethodException | SecurityException | InvocationTargetException ex) {
+                throw new InvalidConfigException("failed to load configured IdentityManager: " + cname, ex);
+            }
         }
-        try {
-            c = Class.forName(cname);
-            Object o = c.newInstance();
-            IdentityManager ret = (IdentityManager) o;
-            log.debug("IdentityManager: " + cname);
-            return ret;
-        } catch (Throwable t) {
-            if (!defaultImplClass.equals(cname) || c != null) {
-                log.error("failed to load configured IdentityManager: " + cname, t);
+        // default
+        return new NoOpIdentityManager();
+    }
+
+    // backwards compat
+    private static Authenticator getAuthenticator(IdentityManager im) {
+        String cname = System.getProperty(Authenticator.class.getName());
+        if (cname != null && im instanceof NoOpIdentityManager) {
```

Comment:
Why the check for instanceof NoOpIdentityManager?

Reply:
this is a fall back in case they have an old-style Authenticator configured via system prop, but I only wanted to do that if they haven't configured an IdentityManager. If for some reason someone has both configured, it should stick with the IdentityManager.... this will get removed once we get passed any bumpy release.

Reply:
I only want this backwards compat to come into play if they have not configured an IdentityManager and have not explicitly configured the NoOpIdentityManager.

Rubin configures their old Authenticator using a java system prop, so this allows that to keep working. They'll have other issues with the IdentityManager interface change, but that can't be avoided without a lot of complexity